bindings/rust/aws-lc-sys-template/build.rs

```diff
@@ -0,0 +1,318 @@
+        if cc_version.contains(REQUIRED_CLANG_VERSION) {
+            assert!(
+                version(cxx).contains(REQUIRED_CLANG_VERSION),
+                "mismatched versions of cc and c++"
+            );
+            return (cc, cxx);
+        } else if cc == "cc" {
+            panic!(
+                "unsupported clang version \"{}\": FIPS requires clang {}",
+                cc_version, REQUIRED_CLANG_VERSION
+            );
+        } else if !cc_version.is_empty() {
+            eprintln!(
+                "warning: FIPS requires clang version {}, skipping incompatible version \"{}\"",
+                REQUIRED_CLANG_VERSION, cc_version
+            );
+        }
+    }
+    unreachable!()
+}
+
+fn test_cmake_command(executable: &str) -> bool {
+    if let Ok(output) = Command::new(executable).arg("--version").output() {
+        return output.status.success();
+    }
+    false
+}
+
+fn prepare_cmake_build(build_fips: bool, build_prefix: Option<&str>) -> cmake::Config {
+    if test_cmake_command("cmake3") {
+        env::set_var("CMAKE", "cmake3");
+    } else {
+        env::set_var("CMAKE", "cmake");
+    }
+
+    let mut cmake_cfg = get_cmake_config();
+
+    if build_fips {
+        let (clang, clangxx) = verify_fips_clang_version();
+        cmake_cfg.define("CMAKE_C_COMPILER", clang);
+        cmake_cfg.define("CMAKE_CXX_COMPILER", clangxx);
+        cmake_cfg.define("CMAKE_ASM_COMPILER", clang);
+        cmake_cfg.define("FIPS", "1");
+    }
+
+    if let Some(symbol_prefix) = build_prefix {
+        cmake_cfg.define("BORINGSSL_PREFIX", symbol_prefix);
+        let pwd = env::current_dir().unwrap();
+        let include_path = pwd.join(AWS_LC_PATH).join("include");
+        cmake_cfg.define(
+            "BORINGSSL_PREFIX_HEADERS",
+            include_path.display().to_string(),
+        );
+    }
+
+    cmake_cfg
+}
+
+fn build_aws_lc() -> PathBuf {
+    let mut cmake_cfg = prepare_cmake_build(false, Some(&prefix_string()));
+
+    cmake_cfg.build_target("ssl").build()
+}
+
+fn main() -> Result<(), String> {
+    use crate::OutputLib::{Crypto, Ssl};
+    use crate::OutputLibType::Static;
+
+    let out_dir = env::current_dir().unwrap();
+    let out_dir = Path::new(&out_dir);
+    let prefix = prefix_string();
+
+    let bindings_file = out_dir.join("src").join("bindings.rs");
+
+    let builder = prepare_bindings_builder(out_dir, Some(&prefix));
+    let bindings = builder.generate().expect("Unable to generate bindings.");
+    bindings
+        .write_to_file(&bindings_file)
+        .expect("Unable to write bindings to file.");
+    modify_bindings(&bindings_file, &prefix).map_err(|err| err.to_string())?;
+
+    let aws_lc_dir = build_aws_lc();
+
+    let libcrypto_dir = Crypto.locate_dir(&aws_lc_dir);
+    let libssl_dir = Ssl.locate_dir(&aws_lc_dir);
+    println!("cargo:rustc-link-search=native={}", libcrypto_dir.display());
+    println!("cargo:rustc-link-search=native={}", libssl_dir.display());
```

Comment:
You have a good point. Since there are 2 libs (libcrypto and libssl), should there be 2 crates?  Or make "ssl" a feature of the `aws-lc-sys` crate?

The include files for libcrypto and libssl are all in the [same directory](https://github.com/awslabs/aws-lc/tree/main/include/openssl), so we'd need to split those when invoking `rust-bindgen` in a way that's consistent with which lib the corresponding functions are placed in.